File: AFRest/src/main/java/com/tomscz/afrest/commons/SupportedWidgets.java
package com.tomscz.afrest.commons;

/**
 * This {@link Enum} holds supported widget. Widget is input field, label, output field and etc.
 * Based on this {@link Enum} is determined which type of component is used.
 * 
 * @author Martin Tomasek (martin@toms-cz.com)
 * 
 * @since 1.0.0.
 */
public enum SupportedWidgets {

    TEXTFIELD("TEXTFIELD"),
    LABEL("LABEL"),
    NUMBERFIELD("NUMBERFIELD"),
    NUMBERLONGFIELD("NUMBERLONGFIELD"),
    NUMBERDOUBLEFIELD("NUMBERDOUBLEFIELD"),
    DROPDOWNMENU("DROPDOWNMENU"),
    CHECKBOX("CHECKBOX"),
    TEXTAREA("TEXTAREA"),
    OPTION("OPTION"),
    CALENDAR("CALENDAR"),
    PASSWORD("PASSWORD");

    private final String name;

    private SupportedWidgets(String name) {
        this.name = name;
    }

    public boolean equalsName(String otherName) {
        return (otherName == null) ? false : name.equals(otherName);
    }

    public String toString() {
        return name;
    }
    
}

File: AFMobile/AFWindowsPhone/AFWindowsPhone/enums/SupportedWidgets.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AFWindowsPhone.enums
{
    public sealed class SupportedWidgets
    {
        public static readonly SupportedWidgets TEXTFIELD = new SupportedWidgets("TEXTFIELD");
        public static readonly SupportedWidgets PASSWORD = new SupportedWidgets("PASSWORD");
        public static readonly SupportedWidgets NUMBERFIELD = new SupportedWidgets("NUMBERFIELD");
        public static readonly SupportedWidgets NUMBERDOUBLEFIELD = new SupportedWidgets("NUMBERDOUBLEFIELD");
        public static readonly SupportedWidgets CALENDAR = new SupportedWidgets("CALENDAR");
        public static readonly SupportedWidgets OPTION = new SupportedWidgets("OPTION");
        public static readonly SupportedWidgets DROPDOWNMENU = new SupportedWidgets("DROPDOWNMENU");
        public static readonly SupportedWidgets CHECKBOX = new SupportedWidgets("CHECKBOX");

        private String widgetName;

        SupportedWidgets(String widgetType)
        {
            this.widgetName = widgetType;
        }

        public String getWidgetName()
        {
            return widgetName;
        }

        internal bool equals(SupportedWidgets tEXTFIELD)
        {
            throw new NotImplementedException();
        }
    }
}
